src/sentry/conf/server.py

```diff
@@ -889,6 +889,9 @@ def SOCIAL_AUTH_DEFAULT_USERNAME() -> str:
     Queue("events.save_event", routing_key="events.save_event"),
     Queue("events.save_event_highcpu", routing_key="events.save_event_highcpu"),
     Queue("events.save_event_transaction", routing_key="events.save_event_transaction"),
+    Queue("events.save_event_transaction_1", routing_key="events.save_event_transaction"),
+    Queue("events.save_event_transaction_2", routing_key="events.save_event_transaction"),
+    Queue("events.save_event_transaction_3", routing_key="events.save_event_transaction"),
```

Comment:
Should the queues have different routing keys? My understanding of AMQP queues is that queues are bound to an exchange with a routing key. Any [message published on an exchange with a matching routing key is delivered to all queues with the same routing key](https://www.rabbitmq.com/tutorials/tutorial-four-python#multiple-bindings). If all of these queues have the same routing key, each of them would receive a copy of messages published with `events.save_event_transaction`

Reply:
That's what I thought initially, though we already have the the post process transactions queue configured this way in prod https://github.com/getsentry/ops/blob/master/k8s/services/getsentry/templates/k8s_regionsilo_ops.py.j2#L13-L17
with the same routing_key 
```
    Queue("post_process_transactions_1", routing_key="post_process_transactions"),
    Queue("post_process_transactions_2", routing_key="post_process_transactions"),
    Queue("post_process_transactions_3", routing_key="post_process_transactions"),
```

They are definitively not getting duplicated messages as they all get 1/3 of the original load. 
I try to run this locally and I observed the load being routed in round robin. 
I think it may be for two reasons:
1. in post process queue we do not specify the queue in the task but in the `apply_async` call. 
2. For the save event, this may be working because of the router implementation `SplitQueueRouter`. 

@wedamija  I see the previous split queue implementation did not even specify the routing key on the split queues. https://github.com/getsentry/sentry/pull/39634/files#diff-866fbc2da71a688b0a929236b2c864dfa8935d4a6423321d2d97a28337c6e6e8L693 Do you know if there was a reason for that?

Reply:
@fpacifici All I really know about this is that we removed the split queues because they were totally unused.

I think that the reason this works is because there's a custom router here: https://github.com/getsentry/ops/blob/c9ff1d47d20a07a004e31907e10b71146fa28e97/k8s/services/getsentry/templates/k8s_regionsilo_ops.py.j2#L20-L26.

That's used here: https://github.com/getsentry/sentry/blob/8c7b54ca2bfa9383ba5f969bfe0dd86485ff67c7/src/sentry/eventstream/base.py#L111-L113

I think this router must override the routing key - although that kind of makes me wonder if the routing key is necessary at all here. 

I'm also not sure why we need to add these 3 queues to the default config - shouldn't this be configured specifically in production configs?

Reply:
@wedamija  what about doing like the previous implementation where we provide a parameter and the queues are automatically generated ?

Reply:
I figured out why it worked (thanks @beezz ).
```
root@rabbitmq-default-0.c.internal-sentry.internal:~# rabbitmqctl list_bindings --vhost /getsentry | grep post_process_tra
        exchange        post_process_transactions_1     queue   post_process_transactions_1     []
        exchange        post_process_transactions       queue   post_process_transactions       []
        exchange        post_process_transactions_2     queue   post_process_transactions_2     []
        exchange        post_process_transactions_3     queue   post_process_transactions_3     []
```

This queue uses the default exchange, not the direct exchange. So the queue is bound to a binding_key with the same name as the queue when it is created.  The direct exchange would indeed fan the message out to all queues.

In fact I tried to produce a message with `binding_key=post_process_transactions` rather than by specifying the queue and the message went nowhere. That, to my understanding, is because the excahnge we use does not have a binding_key = `post_process_transactions`. 

Anyway I will change the binding key to avoid disasters should we change the exchange.

Reply:
> I'm also not sure why we need to add these 3 queues to the default config - shouldn't this be configured specifically in production configs?

I changed the approach. We do not declare all the queues one by one anymore. We have an additional config setting that generates the queues we need. This will be set up in prod only.